plugins/inventory/icinga2.py

```diff
@@ -0,0 +1,217 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2016 Guido Günther <agx@sigxcpu.org>, Daniel Lobato Garcia <dlobatog@redhat.com>
+# Copyright (c) 2018 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: icinga2
+    short_description: Icinga2 inventory source
+    author:
+        - Cliff Hults (@BongoEADGC6) <cliff.hults@gmail.com>
+    requirements:
+        - requests >= 1.1
+    description:
+        - Get inventory hosts from the Icinga2 API.
+        - "Uses a configuration file as an inventory source, it must end in C(.icinga2.yml) or C(.icinga2.yaml)"
+    options:
+      plugin:
+          description: Name of the plugin
+          required: true
+          choices: ['community.general.icinga2']
+      url:
+        description: Root URL of Icinga2 API 
+        required: true
+      user:
+        description: API username
+        required: true
+      password:
+        description: API password
+        required: true
+      validate_certs:
+        description: Enable SSL certificate verification
+        required: true
+'''
+EXAMPLES = '''
+# my.icinga2.yml
+plugin: community.general.icinga2
+url: http://localhost:5665
+user: ansible
+password: secure
+validate_certs: false
+'''
+import re
+import json
+
+from ansible.module_utils.common._collections_compat import MutableMapping
+from distutils.version import LooseVersion
+
+from ansible.errors import AnsibleError, AnsibleParserError
+from ansible.plugins.inventory import BaseInventoryPlugin, Constructable, Cacheable
+from ansible.module_utils.six.moves.urllib.parse import urlencode
+
+# 3rd party imports
+try:
+    import requests
```

Comment:
Why not use Ansible's built-in open_url from https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/urls.py#L1522 ? That would make this plugin also work if requests isn't around.